### PR TITLE
enhancement #771: Improved and merged endpoints for captain changing:

### LINKED ISSUE #771: Improve and merge endpoints for captain changing
As we can see, we have two endpoints for changing captain: the first is common, the second is administrative. They differ only in permission, so take advantage of @JessFreak improvement and **merge endpoints into one with two permissions**.

image

Next task: Make previous captain always be given **STUDENT group role** (if he was in the group).
Don't forget about **Swagger** documentation

### PR DESCRIPTION
**Pull request by JessFreak**
- deleted transferCaptain in GroupController
- moved permission to switchCaptain
- created changeRole in UserService
- deleted one test
- changed Swagger documentation

closes #771

### CODE REVIEW

File: src/v2/api/services/GroupService.ts [hunk 1]
```diff
@@ -304,55 +282,13 @@ export class GroupService {
   async switchCaptain (groupId: string, studentId: string) {
     const isStudentInGroup = await this.isStudentInGroup(groupId, studentId);
     if (!isStudentInGroup) throw new NoPermissionException();
-    const { id: oldCaptainId } = await this.getCaptain(groupId);
+    const { id: oldCaptainId } = await this.getCaptain(groupId) || { id: null };
```
Comment (fokaaas) [inline src/v2/api/services/GroupService.ts:285]: it would be better to just remove destructuring

File: src/v2/api/services/UserService.ts [hunk 2]
```diff
@@ -150,6 +150,33 @@ export class UserService {
     });
   }
 
+  async changeRole (studentId: string, name: RoleName) {
+    const userRole = await this.getGroupRoleDB(studentId);
```
Comment (fokaaas) [inline src/v2/api/services/UserService.ts:154]: wrong naming, it's group role

**Review by fokaaas — CHANGES_REQUESTED** [review]
Also, your endpoint should return updated captain

**Review by fokaaas — APPROVED** [review]

File: src/v2/api/controllers/GroupController.ts [hunk 3]
```diff
@@ -469,38 +473,21 @@ export class GroupController {
...
-    UnauthorizedException:
-      Unauthorized`,
-  })
-  @ApiForbiddenResponse({
-    description: `\n
-    NoPermissionException:
-      You do not have permission to perform this action`,
+  @ApiEndpoint({
+    summary: 'Make a student a captain',
+    permissions: [PERMISSION.GROUPS_CAPTAIN_SWITCH, PERMISSION.GROUPS_$GROUPID_TRANSFER],
```
Comment (illiakroshka) [inline src/v2/api/controllers/GroupController.ts:483]: Endpoint must work when user has 1 of these 2 permissions. But it works only when user has 2 permissions

**Review by illiakroshka — APPROVED** [review]

**Pull request merged**